### PR TITLE
Make exprt::is_zero() support bitfields

### PR DESCRIPTION
**Pull request by chrisr-diffblue**

### CODE REVIEW

**Review by smowton — APPROVED** [review]

**Review by smowton — COMMENTED** [review]
@chrisr-diffblue add the new unit test to the Makefile (only cmake picks it up automatically)

**Comment by chrisr-diffblue** [comment]
Ah! Forgot about the Makefile, sigh. I love having two build systems :-)

File: unit/util/expr.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+SCENARIO("bitfield-expr-is-zero", "[core][util][expr]")
+{
+  GIVEN("An exprt representing a bitfield constant of 3")
+  {
+    const exprt bitfield3 = from_integer(mp_integer(3), c_bit_field_typet(signedbv_typet(32), 4));
+
+    THEN("is_zero() should be false")
+    {
+      REQUIRE(!bitfield3.is_zero());
```
Comment (owen-mc-diffblue) [inline unit/util/expr.cpp]: `REQUIRE_FALSE` is slightly easier to understand

**Review by owen-mc-diffblue — APPROVED** [review]

File: unit/util/expr.cpp (hunk 1 above)
Comment (chrisr-diffblue) [reply unit/util/expr.cpp]: Yeah, that's nicer - I'll change.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 38c9cb4c30ddaf191b163f708963ac56525bc7c1).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80001408
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 54cc8187f6bd96b9faea8a707fb60c694f5e269a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80004998

**Pull request merged**